docs/dsl/headers.md

```diff
@@ -3,20 +3,154 @@ id: headers
 title: Headers
 ---
 
-**ZIO HTTP** provides support for all HTTP headers (as defined
-in [RFC2616](https://datatracker.ietf.org/doc/html/rfc2616) ) along with custom headers.
+**ZIO HTTP** provides support for all HTTP headers (as defined in [RFC2616](https://datatracker.ietf.org/doc/html/rfc2616)) along with custom headers.
+
+In ZIO HTTP we have two related types of headers:
+
+- `Header` represents a single HTTP header
+- `Headers` represents an immutable collection of headers
+
+## Header
+
+We can think of `Header` as a key-value pair, where the key is the header name and the value is the header value:
+
+```scala
```

Comment:
I think this is misleading. The `renderedValue` is a string representation of the header. But the real value is defined by `headerType`.

Reply:
Corrected!